File: ios/Natives/New Group/LocaleModule.m
//
//  LocaleModule.m
//  Walletly
//
//  Created by Gabrielle on 27/07/25.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(LocaleModule, NSObject)

RCT_EXTERN_METHOD(getDeviceLanguage:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/Natives/Locale/LocaleModule.swift
import Foundation
import React

@objc(LocaleModule)
class LocaleModule: NSObject {

  @objc
  func getDeviceLanguage(_ resolve: @escaping RCTPromiseResolveBlock,
                         rejecter reject: @escaping RCTPromiseRejectBlock) {
    let locale = Locale.current
    let result: NSDictionary = ["language": locale.identifier]
    resolve(result)
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
